packages/jira-adapter/src/filters/script_runner/workflow_scripts_dc.ts

```diff
@@ -0,0 +1,146 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { logger } from '@salto-io/logging'
+import { safeJsonStringify, WalkOnFunc, walkOnValue, WALK_NEXT_STEP } from '@salto-io/adapter-utils'
+import { isInstanceElement, Element, isInstanceChange, isAdditionOrModificationChange, getChangeData, Value } from '@salto-io/adapter-api'
+import { FilterCreator } from '../../filter'
+import { WORKFLOW_TYPE_NAME } from '../../constants'
+
+const log = logger(module)
+const SCRIPT_RUNNER_DC_TYPES = ['com.onresolve.jira.groovy.GroovyFunctionPlugin',
+  'com.onresolve.jira.groovy.GroovyValidator',
+  'com.onresolve.jira.groovy.GroovyCondition']
+const DC_BASE64_PREFIX = 'YCFg'
+const CANNED_SCRIPT = '"canned-script"'
+const FIELD_COMMENT_TYPE = 'com.onresolve.scriptrunner.canned.jira.workflow.postfunctions.CommentIssue'
+
+
+const decodeBase64 = (base64: string): string => {
+  if (!base64.startsWith(DC_BASE64_PREFIX)) {
+    log.warn(`Could not decode DC ScriptRunner script, expected base64, got: ${base64}`)
+    return base64
+  }
+  return Buffer.from(base64.substring(DC_BASE64_PREFIX.length), 'base64').toString('utf8')
+}
+
+const encodeBase64 = (script: string): string => DC_BASE64_PREFIX + Buffer.from(script).toString('base64')
```

Comment:
I think it's better to do 
```suggestion
const encodeBase64 = (script: string): string => Buffer.from('`!`' + script).toString('base64')
```

Concatenating base64 parts like that feels dangerous

Reply:
As it is the start and exactly 4 base64 chars it seems ok to me and cleaner this way, do you have a specific scenario in mind?

Reply:
I'm not sure it is cleaner, the encoded part does not have any meaning so it's not clear what you are actually adding to the string. I needed to decode the string to understand what's happening in this part